### PR TITLE
Integrating NMRium into ELN

### PR DESCRIPTION
**Pull request by baolanlequang**
- [x] rather 1-story 1-commit than sub-atomic commits

- [x] commit title is meaningful =>  git history search

- [x] commit description is helpful => helps the reviewer to understand the changes

- [ ] code is up-to-date with the latest developments of the target branch (rebased to it or whatever) => :fast_forward:-merge for linear history is favoured

- [x] added code is linted

- [x] tests are passing (at least locally): we still have some random test failure on CI. thinking of asking spec/examples.txt to be commited

- [x] in case the changes are visible to the end-user,  video or screenshots should be added to the PR => helps  with user testing

- [x] testing coverage improvement is improved.

- [x] CHANGELOG :  add a bullet point on top (optional: reference to github issue/PR )

- [x] parallele PR for documentation  on docusaurus  if the feature/fix is tagged for a release: see https://github.com/ComPlat/chemotion_saurus/pull/141

### CODE REVIEW

**Comment by JanCBrammer** [comment]
@baolanlequang, how does this PR relate to #1018, and #965?

**Comment by baolanlequang** [comment]
###  This function needs 2 PRs from chem-spectra-app 
https://github.com/ComPlat/chem-spectra-app/pull/55
https://github.com/ComPlat/chem-spectra-app/pull/57

### Document of how to use it on Chemosaurus: 
https://github.com/ComPlat/chemotion_saurus/pull/141

### Demo video:

https://user-images.githubusercontent.com/9006482/205292349-ebf5da65-febb-414f-98cf-3494b3c54854.mp4

**Comment by baolanlequang** [comment]
This PR requires https://github.com/ComPlat/chemotion_ELN/pull/1104 to work properly

File: app/packs/src/components/nmriumWrapper/NMRiumDisplayer.js [hunk 1]
```diff
@@ -100,7 +100,13 @@ export default class NMRiumDisplayer extends React.Component {
...
         if (nmrWrapperActionType !== '') {
           const nmriumData = eventData.data;
-          this.setState({ nmriumData });
+          const { version } = nmriumData;
+          if (version > 3) {
+            this.setState({ nmriumData: nmriumData.data });
+          }
+          else {
+            this.setState({ nmriumData });
+          }
```
Comment (ptrxyz) [inline app/packs/src/components/nmriumWrapper/NMRiumDisplayer.js:109]: Can we improve this to act if things go wrong (--> error message would suffice)?
If data version 5, for example, changes the format again, we do not handle it. Users will not get feedback. :/

**Review by PiTrem — APPROVED** [review]

**Pull request merged**